srv/salt/_modules/cephinspector.py

```diff
@@ -2,6 +2,60 @@
 # vim: ts=8 et sw=4 sts=4
 
 import os
+import socket
+from subprocess import Popen, PIPE
+
+def _extract_journal_partition(ceph_disk_str):
+    """
+    From a ceph-disk string of the form: "/dev/vdb2 ceph data, active, cluster ceph, osd.1, journal /dev/vdb1"
+    returns the journal entry "/dev/vdb"
+    """
+    try:
+        journal_partition_index = ceph_disk_str.split().index("journal") + 1
+        return ceph_disk_str.split()[journal_partition_index].rstrip("1234567890")
+    except (ValueError, IndexError) as e:
+        # TODO: anything more drastic needed than to simply omit the journal entry?
+        return None
+
+def _append_to_ceph_disk(ceph_disks, partition, d):
+    """
+    Populate ceph_disks dictionary with data and journal partitions.
+    """
+    try:
+        ceph_disks["ceph"]["storage"]["osds"][partition]
+    except KeyError, e:
+        ceph_disks["ceph"]["storage"]["osds"][partition] = {}
+    finally:
+        ceph_disks["ceph"]["storage"]["osds"][partition]["format"] = "filestore"
+        ceph_disks["ceph"]["storage"]["osds"][partition]["journal"] = _extract_journal_partition(d)
+
+def get_ceph_disks_yml(**kwargs):
+    """
+    Generates yml representation of Ceph filestores on a given node.
+    Returns something like: {"ceph": {"storage": {"osds": {"/dev/foo":
+                                                            {"format": "filestore",
+                                                             "journal": "/dev/bar"}}}}}
+    """
+    ceph_disk_list = Popen("ceph-disk list", stdout=PIPE, stderr=PIPE, shell=True)
```

Comment:
You might be better off with `ceph-disk list --format=json`. It gives you more information, and should hopefully be easier to machine parse than the human-readable output.